### PR TITLE
fix: add function to retrieve rejected report tokens

### PR DESCRIPTION
**Pull request by Ignacio-Freire**

### CODE REVIEW

File: contracts/LosslessReporting.sol [hunk 1]
```diff
@@ -298,4 +298,19 @@ contract LosslessReporting is ILssReporting, Initializable, ContextUpgradeable,
...
     }
 
-}
+    /// @notice This function allows the admin to retrieve tokens that are stuck in the contract
+    /// @param _token Address of the token to be retrieved
+    /// @param _to Address to which the tokens will be sent
+    /// @param _amount Amount of tokens to be retrieved
+    function retrieveRejectedReportTokens(ILERC20 _token, address _to, uint256 _amount) public onlyLosslessAdmin {
+        require(_to != address(0), "LSS: Cannot send to zero address");
+        require(_token != stakingToken, "LSS: Cannot retrieve staking tokens");
```
Comment (DomantasP) [inline contracts/LosslessReporting.sol:307]: The main goal of this function is to retrieve stuck staking tokens.

File: contracts/LosslessReporting.sol [hunk 2]
```diff
@@ -298,4 +298,19 @@ contract LosslessReporting is ILssReporting, Initializable, ContextUpgradeable,
         emit CompensationRetrieve(_adr, _amount);
     }
 
-}
+    /// @notice This function allows the admin to retrieve tokens that are stuck in the contract
+    /// @param _token Address of the token to be retrieved
+    /// @param _to Address to which the tokens will be sent
+    /// @param _amount Amount of tokens to be retrieved
+    function retrieveRejectedReportTokens(ILERC20 _token, address _to, uint256 _amount) public onlyLosslessAdmin {
```
Comment (DomantasP) [inline contracts/LosslessReporting.sol:305]: Function doesn't really check if the report is rejected. We should have some logic that would require that would check that tokens can be retrieved only when latest report is expired.

**Review by DomantasP — CHANGES_REQUESTED** [review]

File: contracts/LosslessReporting.sol (hunk 1 above)
Comment (Ignacio-Freire) [reply contracts/LosslessReporting.sol]: Oops! Leftover from copy&paste [c9e7fd0](https://github.com/Lossless-Cash/lossless-v3/pull/116/commits/c9e7fd0810a1199ed1111f652221109326a01e32)

File: contracts/LosslessReporting.sol (hunk 2 above)
Comment (Ignacio-Freire) [reply contracts/LosslessReporting.sol]: Added check [0a874ca](https://github.com/Lossless-Cash/lossless-v3/pull/116/commits/0a874ca7d46994541403cffad25de32a06039f94)

File: contracts/LosslessReporting.sol [hunk 3]
```diff
@@ -302,11 +302,15 @@ contract LosslessReporting is ILssReporting, Initializable, ContextUpgradeable,
...
+    function retrieveRejectedReportTokens(ILERC20 _token, address _to, uint256 _amount, uint256 _reportId) public onlyLosslessAdmin {
         require(_to != address(0), "LSS: Cannot send to zero address");
-        require(_token != stakingToken, "LSS: Cannot retrieve staking tokens");
         require(_amount > 0, "LSS: Amount must be greater than zero");
 
+        // Check the status of the report
+        Report storage report = reportInfo[_reportId];
+        require(report.status == ReportStatus.Rejected || (report.status == ReportStatus.Expired && block.timestamp > report.reportTimestamps + reportLifetime), "LSS: Report not expired or rejected");
+
         uint256 contractBalance = _token.balanceOf(address(this));
```
Comment (DomantasP) [inline contracts/LosslessReporting.sol:314]: But this allows us to withdraw all the staking tokens when any report is expired. So if report x is still active, but we submit another id which is expired we can drain all the funds. So this should transfer our only the reporting amount.

**Review by DomantasP — CHANGES_REQUESTED** [review]

File: contracts/LosslessReporting.sol (hunk 3 above)
Comment (Ignacio-Freire) [reply contracts/LosslessReporting.sol:317]: Yeah, that's true. My bad I went int with a more broader approach but I understand it needed to be specific now. Did the necessary refactors. [5d8ff3e](https://github.com/Lossless-Cash/lossless-v3/pull/116/commits/5d8ff3ed7d3f01b060364345465f1b87cf69987a)

**Review by DomantasP — APPROVED** [review]

**Pull request merged**